packages/management-api/src/managed-pod.ts

```diff
@@ -117,36 +160,60 @@ export class ManagedPod {
       })
         .done((data: string, textStatus: string, xhr: JQueryXHR) => {
           if (xhr.status !== 200) {
-            reject()
+            this.setManagementError(xhr.status, textStatus)
+            reject(this.getManagementError())
             return
           }
 
+          let jsonResponse
           try {
-            const resp = JSON.parse(data)
-            if ('error' in resp) {
-              throw new Error(resp.error)
-            }
-
-            if ('value' in resp && 'agent' in resp.value) {
-              log.debug('Found jolokia agent at:', this.jolokiaPath, 'version:', resp.value.agent)
-              resolve(this.jolokiaPath)
-              return
-            } else {
-              throw new Error('Detected jolokia but cannot determine agent or version')
-            }
+            jsonResponse = JSON.parse(data)
           } catch (e) {
-            // Parse error should mean redirect to html
-            reject(e)
+            let msg
+            if (e instanceof Error) msg = e.message
+            else msg = String(e)
+
+            this.setManagementError(500, msg)
+            reject(this.getManagementError())
             return
           }
-        })
-        .fail((xhr: JQueryXHR, _: string, error: string) => {
-          if (error) {
-            reject(`Jolokia Probe Error: ${error}`)
+
+          if (!jsonResponse || 'error' in jsonResponse) {
+            this.setManagementError(500, jsonResponse.error)
+            reject(this.getManagementError())
+            return
           }
 
-          reject(`${xhr.status} ${xhr.statusText}`)
+          if ('value' in jsonResponse && 'agent' in jsonResponse.value) {
+            log.debug('Found jolokia agent at:', this.jolokiaPath, 'version:', jsonResponse.value.agent)
+            resolve(this.jolokiaPath)
+            return
+          } else {
+            this.setManagementError(500, 'Detected jolokia but cannot determine agent or version')
+            reject(this.getManagementError())
+            return
+          }
+        })
+        .fail((xhr: JQueryXHR, _: string, error: string) => {
+          const msg = `Jolokia Connect Error (${xhr.status}): ${error ?? xhr.statusText}`
+          this.setManagementError(xhr.status, msg)
+          reject(this.getManagementError())
         })
     })
   }
+
+  search(successCb: () => void, failCb: (error: Error) => void) {
+    this.jolokia.search('org.apache.camel:context=*,type=routes,*', {
```

Comment:
Just in case, what is this method for?  Note a pod may not necessarily contain a Camel context. The only requirement is that a pod should be _hawtio-enabled_ which means it has a Jolokia port, not necessarily to be a Camel app. For example, we should also support Artemis brokers which don't have Camel contexts at all.

Reply:
Ah I see. Thanks for the explanation. It's then totally a different topic, but maybe later we can improve this part (of course, with another pull req than this one) by making it display something like `non camel` or removing the `route` label when it's really not a Camel app.